### ISSUE TITLE
Region rule false negative

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2049]
If `<body>` contains text content, the region rule will pass the entire page even if there are violations. This is because we stop looking at children nodes if the node contains content and we do not evaluate the `<body>` element due to the selector being `"body *"`.

```html
<html>
<body>
  <div>hello</div>
  foobar
</body>
</html>
```

**Comment by straker** [comment]
We should ignore text nodes that are direct children of body since we cannot target the text node itself and saying "body must be contained with in a landmark" doesn't make sense.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-coconut(4.3.0.05015), only `div` throwing a voilation of "Some page content is not contained by landmarks" (attached screenshot), not for `body` element.  


![image](https://user-images.githubusercontent.com/47409579/74848217-78161b00-535d-11ea-8c34-20a6f12b61fb.png)

**Comment by jeankaplansky** [comment]
release notes for #2049

**Comment by straker** [comment]
No docs needed
